# Intraocular Bacillus genomic pipeline  
### Pipeline summary  
The Intraocular Bacillus genomic pipeline generates, for each sample:
1. Examining	quality	of reads
2. De novo sequence assemble
3. Assembly evaluate
4. Genome annotation
5. Phylogenetic and clustering analysis
6. The pan genome analysis

### Pipeline components
* trim read: [skewer 0.2.2](https://github.com/relipmoc/skewer)
* quality control: [fastqc](https://github.com/s-andrews/FastQC)
* Genome assembly: [KmerGenie 1.7051](http://kmergenie.bx.psu.edu/) and [SPAdes 3.12.0](http://cab.spbu.ru/software/spades/)
* Genome annotation: [Prokka 1.13](https://github.com/tseemann/prokka)
* Phylogenetic analysis: [RAxML 8.2.12](https://github.com/stamatak/standard-RAxML)
* Pan genome analysis: [Roary 3.11.2](https://github.com/sanger-pathogens/Roary) and [Scoary 1.6.16](https://github.com/AdmiralenOla/Scoary)

### Pipeline parameters
This section contains the full list of inputs and parameters provided to each method.
The following variables must be defined:
* `star_index`: path to the directory containing the STAR index
* `fastq1` and `fastq2`: paths to the two FASTQ files
* `sample_id`: sample identifier; this will be prepended to output files
* `rsem_reference`: path to the directory containing the RSEM reference
* `genome_fasta`: path to the reference genome (`Homo_sapiens_assembly38_noALT_noHLA_noDecoy_ERCC.fasta` as described above)
* `genes_gtf`: path to the collapsed, gene-level GTF (`gencode.v26.GRCh38.ERCC.genes.gtf` as described above)
* `star_bam_file`: name/file path of the BAM generated by the STAR aligner, by default `${sample_id}.Aligned.sortedByCoord.out.bam`.
* `md_bam_file`: name of the BAM generated by Picard MarkDuplicates.
1. skewer
    ```bash
    skewer ${fastq1} ${fastq2}\
        -m pe\
        -f sanger\
        -r 0.1\
        -d 0.03\
        -q 20\
        -Q 20\
        -l 75\
        -n\ 
        -L 150\ 
        -e\ 
        -o ${sample_id}\
        --quiet\
        --compress\
        -X\
        -t 4
    ```
    2. 
